### ISSUE TITLE
Set up hosted instance for devs

### ISSUE THREAD

**Issue by petecooper** [textpattern/unit-test#2]
NTS

**Comment by Bloke** [comment]
Nice, thank you. This is interesting as it's multifaceted:

* Some base Txp installation, perhaps bleeding edge.
* Merge this repo with the above codebase, so the theme is accessible to the site.
* Import the well-known DB, and merge any files.
* Trigger "log in" event to run any upgrade?
* Run tests via curl and report somehow if they fail, without spamming people who have acknowledged receipt.
* Permit manual login so people can run their own tests.
* Somehow prevent manual tinkering of tests from affecting periodic automated runs until they are stamped as "ready".
* Periodically replace the DB and content with a new baseline as things change. Ostensibly on release of each new version. The test framework then compares last version with any dev changes 

The above might well contain incompatible goals and the automated test environment needs to be separate from the manual one. Not sure yet. Would be nice to run it all from one place but unsure of the mechanics and general level of script-fu required to get it singing.

**Comment by petecooper** [comment]
Revisiting this since it got a bit lost…

Anything to add to https://github.com/textpattern/unit-test/issues/2#issuecomment-801233422 @Bloke, or is that a good jumping off point?

**Comment by Bloke** [comment]
I think the post above encapsulates most of what would be desirable in such an instance.

Not sure how much is doable. But if it's not easily scriptable then I think it damn well should be, so it's an opportunity to tweak core to allow such an environment to be spun up more easily.

**Comment by Bloke** [comment]
Q1: is it possible to plonk a cookie file in an instance and have it log in? Or could we maybe manually log in once, grab the cookie and stick it in this repo. If that's possible, the downside is it expires after 30 days so we would need to find a way to refresh the file periodically. Or hack it so the expiry is never.

**Comment by petecooper** [comment]
Is there scope for a `unit-test` branch that doesn't need a login and is essentially `dev` with patched code? That way we bypass the cookie and it won't need maintaining (also allows others to have a 'turbo debug' version for their own hacking.

**Comment by Bloke** [comment]
Absolutely. The auto login would only be for automatic unit testing that could report if anything broke when the code base was automatically rebuilt with the latest bleeding edge code. If it's only for manual unit testing then yeah, patched code base with all the unit tests we can design would be awesome.
